### ISSUE TITLE
ImportError cannot import name 'Buffer' from 'collections.abc'

### ISSUE THREAD

**Issue by JiuWeiA** [numpy/numpy#25020]
### Steps to reproduce:

`import numpy
import cv2
import pyscreeze
import pyscreenshot
import pyautogui

im = pyautogui.screenshot()
def show(img):
    ''' 显示一个图片 '''
    cv2.imshow('image', img)
    cv2.waitKey(0)
    cv2.destroyAllWindows()

show(im)`

that was my code,and some wrong happend,I try to solve it but I can't   the logs said  ImportError: cannot import name 'Buffer' from 'collections.abc'   How can I solve it? thanks

### Error message:

```shell
`(pythonProject) PS D:\西南大学\Python\程序\pythonProject> python -c "import numpy, sys; sys.exit(numpy.test() is False)"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "D:\Program Files\PyCharm Community Edition 2023.2.3\pythonProject\Lib\site-packages\numpy\__init__.py", line 163, in <module>
    from . import lib
  File "D:\Program Files\PyCharm Community Edition 2023.2.3\pythonProject\Lib\site-packages\numpy\lib\__init__.py", line 23, in <module>
    from . import index_tricks
  File "D:\Program Files\PyCharm Community Edition 2023.2.3\pythonProject\Lib\site-packages\numpy\lib\index_tricks.py", line 12, in <module>
    import numpy.matrixlib as matrixlib
  File "D:\Program Files\PyCharm Community Edition 2023.2.3\pythonProject\Lib\site-packages\numpy\matrixlib\__init__.py", line 4, in <module>
    from . import defmatrix
  File "D:\Program Files\PyCharm Community Edition 2023.2.3\pythonProject\Lib\site-packages\numpy\matrixlib\defmatrix.py", line 12, in <module>
    from numpy.linalg import matrix_power
  File "D:\Program Files\PyCharm Community Edition 2023.2.3\pythonProject\Lib\site-packages\numpy\linalg\__init__.py", line 73, in <module>
    from . import linalg
  File "D:\Program Files\PyCharm Community Edition 2023.2.3\pythonProject\Lib\site-packages\numpy\linalg\linalg.py", line 37, in <module>
    from numpy._typing import NDArray
  File "D:\Program Files\PyCharm Community Edition 2023.2.3\pythonProject\Lib\site-packages\numpy\_typing\__init__.py", line 182, in <module>
    from ._array_like import (
  File "D:\Program Files\PyCharm Community Edition 2023.2.3\pythonProject\Lib\site-packages\numpy\_typing\_array_like.py", line 82, in <module>
    from collections.abc import Buffer
ImportError: cannot import name 'Buffer' from 'collections.abc' (C:\Users\峰久\AppData\Local\Programs\Python\Python312\Lib\collections\abc.py)
`
```

**Comment by BvB93** [comment]
What Python version are you using here (the `sys.version_info` output as per the issue template)? `collections.abc.Buffer` was added in to the standard library in Python 3.12, so if you're supposedly on this version then its absence implies there's something strange going on with your environment,

**Comment by JiuWeiA** [comment]
> What Python version are you using here (the output as per the issue template)? was added in to the standard library in Python 3.12, so if you're supposedly on this version then its absence implies there's something strange going on with your environment,`sys.version_info``collections.abc.Buffer`

YES I use the lastest vision of python in the python's webpage  3.12   should I downgrade my Python to use?

**Comment by BvB93** [comment]
I'd give it a shot, yes. With the currently provided info all I can tell you that *something* is broken regarding your python 3.12 environment, which is not something we can do an awful lot about.

**Comment by JiuWeiA** [comment]
> I'd give it a shot, yes. With the currently provided info all I can tell you that _something_ is broken regarding your python 3.12 environment, which is not something we can do an awful lot about.

Thanks,I solved that   after downgrade  python to 3.11.0  Everything is OK
